chutney/jira/src/main/java/com/chutneytesting/jira/domain/JiraXrayService.java

```diff
@@ -52,10 +52,20 @@ public JiraXrayService(JiraRepository jiraRepository, JiraXrayClientFactory jira
         loadJiraServerConfiguration();
     }
 
-    public void updateTestExecution(Long campaignId, Long campaignExecutionId, String scenarioId, ReportForJira report) {
+    public void updateTestExecution(Long campaignId, Long campaignExecutionId, String scenarioId, Optional<String> datasetId, ReportForJira report) {
         JiraXrayApi jiraXrayApi = createHttpJiraXrayImpl();
+        String testKey;
 
-        String testKey = jiraRepository.getByScenarioId(scenarioId);
+        Optional<Map<String, String>> scenarioLinkedDataset = Optional.ofNullable(jiraRepository.getAllLinkedScenariosWithDataset().get(scenarioId));
+        boolean isDatasetLinkAvailable = datasetId.isPresent() &&
+            scenarioLinkedDataset.isPresent() &&
+            scenarioLinkedDataset.get().containsKey(datasetId.get());
+
+        if (isDatasetLinkAvailable) {
+            testKey = scenarioLinkedDataset.get().getOrDefault(datasetId.get(), "");
+        } else {
+            testKey = jiraRepository.getByScenarioId(scenarioId);
+        }
```

Comment:
is same as : 
     
```
       String testKey = "";

        if (datasetId.isPresent()) {
            Map<String, String> scenarioLinkedDataset = jiraRepository.getAllLinkedScenariosWithDataset().get(scenarioId);
            testKey = scenarioLinkedDataset.getOrDefault(datasetId.get(), jiraRepository.getByScenarioId(scenarioId));
        }
```

Reply:
Not exactely the same.
jiraRepository.getAllLinkedScenariosWithDataset().get(scenarioId) can be null